src/Compilers/CSharp/Test/Semantic/Semantics/QueryTests.cs

```diff
@@ -2761,44 +2770,65 @@ static void Main()
 }
 ";
             var expectedDiagnostics = new DiagnosticDescription[] {
-                // CS0718: 'GC': static types cannot be used as type arguments
+                // file.cs(9,18): error CS0718: 'GC': static types cannot be used as type arguments
                 //         var q2 = string.Empty.Cast<GC>().Select(x => x);
                 Diagnostic(ErrorCode.ERR_GenericArgIsStaticClass, "string.Empty.Cast<GC>").WithArguments("System.GC").WithLocation(9, 18),
-                // CS0718: 'GC': static types cannot be used as type arguments
+                // file.cs(9,54): error CS0029: Cannot implicitly convert type 'System.GC' to 'TResult'
```

Comment:
>// file.cs(9,54): error CS0029: Cannot implicitly convert type 'System.GC' to 'TResult' [](start = 16, length = 87)

It looks like three additional errors in this scenario are cascading and the experience would be better if they are not reported.